go/acct/net_flow.go

```diff
@@ -0,0 +1,151 @@
+	return string(sqlBytes)
+}
+
+type NetFlowBandwidthAccountingRecs []NetFlowBandwidthAccountingRec
+
+func (array *NetFlowBandwidthAccountingRecs) Append(ip string, TimeBucket time.Time) int {
+	index := len(*array)
+	*array = append(*array, NetFlowBandwidthAccountingRec{Ip: ip, TimeBucket: TimeBucket, UniqueSession: "NETFLOW-" + ip})
+	return index
+}
+
+func (array NetFlowBandwidthAccountingRecs) ToSQL() string {
+	if len(array) == 0 {
+		return ""
+	}
+
+	sql :=
+		`INSERT INTO bandwidth_accounting (tenant_id, unique_session_id, mac, time_bucket, in_bytes, out_bytes)
+    SELECT * FROM (
+        SELECT tenant_id, unique_session_id, mac, time_bucket, in_bytes_, out_bytes_ FROM (
+            `
+	first := array[0]
+	sql += first.ToSQLSelect()
+	for _, rec := range array[1:] {
+		sql += "            UNION ALL " + rec.ToSQLSelect()
+	}
+
+	sql += `        ) as time_buckets INNER JOIN ip4log as ip4 ON time_buckets.ip = ip4.ip
+    ) as x
+ON DUPLICATE KEY UPDATE in_bytes = in_bytes + VALUES(in_bytes), out_bytes = out_bytes + VALUES(out_bytes);`
+
+	return sql
+}
+
+func (h *PfAcct) HandleFlows(header *netflow5.Header, flows []netflow5.Flow) {
+	recs := h.NetFlowV5ToBandwidthAccounting(header, flows)
+	sql := recs.ToSQL()
+	if sql != "" {
+		h.Db.Exec(sql)
+	}
+}
+
+func (h *PfAcct) IpAddressAllowed(ip net.IP) bool {
+	if h.AllNetworks || len(h.AllowedNetworks) == 0 {
+		return true
+	}
+
+	for _, n := range h.AllowedNetworks {
+		if n.Contains(ip) {
+			return true
+		}
+	}
+
+	return false
+}
+
+func (h *PfAcct) NetFlowV5ToBandwidthAccounting(header *netflow5.Header, flows []netflow5.Flow) NetFlowBandwidthAccountingRecs {
+	recs := NetFlowBandwidthAccountingRecs{}
+	lookup := map[string]int{}
+	srcIndex := 0
+	dstIndex := 0
+	unixTime := time.Unix(int64(header.UnixSecs()), int64(header.UnixNsecs()))
+	unixTime = unixTime.Truncate(h.TimeDuration)
+	for _, flow := range flows {
+		srcIndex = -1
+		dstIndex = -1
+		var found bool
+		srcIP := flow.SrcIP()
+		if h.IpAddressAllowed(srcIP) {
+			srcIpStr := srcIP.String()
+			if srcIndex, found = lookup[srcIpStr]; !found {
+				srcIndex = recs.Append(srcIpStr, unixTime)
+				lookup[srcIpStr] = srcIndex
+			}
+		}
+
+		dstIP := flow.DstIP()
+		if h.IpAddressAllowed(dstIP) {
+			dstIpStr := dstIP.String()
+			if dstIndex, found = lookup[dstIpStr]; !found {
+				dstIndex = recs.Append(dstIpStr, unixTime)
+				lookup[dstIpStr] = dstIndex
+			}
+		}
+
+		layer3Bytes := uint64(flow.DOctets())
+		if srcIndex != -1 {
+			recs[srcIndex].OutBytes += layer3Bytes
+		}
+
+		if dstIndex != -1 {
+			recs[dstIndex].InBytes += layer3Bytes
+		}
+
+	}
+
+	return recs
+}
+
+func (h *PfAcct) netflowProcessor() (*processor.Processor, error) {
+	addr := "127.0.0.1:" + h.NetFlowPort
+	conn, err := net.ListenPacket("udp", addr)
+	if err != nil {
+		return nil, err
+	}
+
+	log.LoggerWContext(context.Background()).Info("Starting listening to netflow at '" + addr + "'")
```

Comment:
You should use a global context with a logger in it, otherwise you'll be creating a new logger everytime